sklearn/metrics/tests/test_classification.py

```diff
@@ -739,31 +736,165 @@ def test_likelihood_ratios_errors(params, err_msg):
 def test_likelihood_ratios():
     # Build confusion matrix with tn=9, fp=8, fn=1, tp=2,
     # sensitivity=2/3, specificity=9/17, prevalence=3/20,
     # LR+=34/24, LR-=17/27
     y_true = np.array([1] * 3 + [0] * 17)
     y_pred = np.array([1] * 2 + [0] * 10 + [1] * 8)
 
-    pos, neg = class_likelihood_ratios(y_true, y_pred)
+    pos, neg = class_likelihood_ratios(y_true, y_pred, replace_undefined_by=np.nan)
     assert_allclose(pos, 34 / 24)
     assert_allclose(neg, 17 / 27)
 
     # Build limit case with y_pred = y_true
-    pos, neg = class_likelihood_ratios(y_true, y_true)
+    pos, neg = class_likelihood_ratios(y_true, y_true, replace_undefined_by=np.nan)
+    # TODO(1.9): replace next line with `assert_array_equal(pos, 1.0)`, since
+    # `replace_undefined_by` has a new default:
     assert_array_equal(pos, np.nan * 2)
     assert_allclose(neg, np.zeros(2), rtol=1e-12)
 
     # Ignore last 5 samples to get tn=9, fp=3, fn=1, tp=2,
     # sensitivity=2/3, specificity=9/12, prevalence=3/20,
     # LR+=24/9, LR-=12/27
     sample_weight = np.array([1.0] * 15 + [0.0] * 5)
-    pos, neg = class_likelihood_ratios(y_true, y_pred, sample_weight=sample_weight)
+    pos, neg = class_likelihood_ratios(
+        y_true, y_pred, sample_weight=sample_weight, replace_undefined_by=np.nan
+    )
     assert_allclose(pos, 24 / 9)
     assert_allclose(neg, 12 / 27)
 
 
+# TODO(1.9): remove test
+@pytest.mark.parametrize("raise_warning", [True, False])
+def test_likelihood_ratios_raise_warning_deprecation(raise_warning):
+    """Test that class_likelihood_ratios raises a `FutureWarning` when `raise_warning`
+    param is set."""
+    y_true = np.array([1, 0])
+    y_pred = np.array([1, 0])
+
+    msg = "`raise_warning` was deprecated in version 1.7 and will be removed in 1.9."
+    with pytest.warns(FutureWarning, match=msg):
+        class_likelihood_ratios(y_true, y_pred, raise_warning=raise_warning)
+
+
+# TODO(1.9): remove test
+def test_likelihood_ratios_raise_default_deprecation():
+    """Test that class_likelihood_ratios raises a `FutureWarning` when `raise_warning`
+    and `replace_undefined_by` are both default."""
+    y_true = np.array([1, 0])
+    y_pred = np.array([1, 0])
+
+    msg = "The default return value of `class_likelihood_ratios` in case of a"
+    with pytest.warns(FutureWarning, match=msg):
+        class_likelihood_ratios(y_true, y_pred)
+
+
+def test_likelihood_ratios_replace_undefined_by_worst():
+    """Test that class_likelihood_ratios returns the worst scores `1.0` for both LR+ and
+    LR- when `replace_undefined_by="worst"` is set."""
+    # This data causes fp=0 (0 false positives) in the confusion_matrix and a division
+    # by zero that affects the positive_likelihood_ratio:
+    y_true = np.array([1, 1, 0])
+    y_pred = np.array([1, 0, 0])
+
+    positive_likelihood_ratio, _ = class_likelihood_ratios(
+        y_true, y_pred, replace_undefined_by="worst"
+    )
+    assert positive_likelihood_ratio == 1.0
```

Comment:
Just to avoid an exact comparison with a floating point, it is better to use `pytest.approx`. It is the equivalent from changing from `assert_array_equal` to `assert_allclose`.

```suggestion
    assert positive_likelihood_ratio == pytest.approx(1.0)
```